src/avro_util.erl

```diff
@@ -62,6 +63,72 @@
 -type validate_fun() :: fun((type_or_name(), validate_hist()) -> ok).
 
 %%%_* APIs =====================================================================
+%% @doc Checks whether two schemas are compatible in sense that an event
+%% encoded with the writer schema can be decoded by the reader schema.
+%% Schema compatibility is defined in
+%% https://avro.apache.org/docs/1.8.1/spec.html#Schema+Resolution
+%% @end
+-spec avro_schema_compatible(avro_type(), avro_type()) ->
+                                true | false.
+avro_schema_compatible(Reader, Writer)
+  when ?IS_RECORD_TYPE(Reader) andalso ?IS_RECORD_TYPE(Writer) ->
+  SameFullName =
+    avro:get_type_fullname(Reader) =:= avro:get_type_fullname(Writer),
+  ReaderTypes = avro_record:get_all_field_data(Reader),
+  WriterTypes = avro_record:get_all_field_data(Writer),
+  FieldsDecodable =
+    lists:all(
+      fun({FieldName, FieldType, Default}) ->
+          case lists:keysearch(FieldName, 1, WriterTypes) of
+            false ->
+              Default =/= ?NO_VALUE;
+            {value, {_, WriterType, _}} ->
+              avro_schema_compatible(FieldType, WriterType)
+          end
+      end,
+      ReaderTypes
+     ),
+  SameFullName andalso FieldsDecodable;
+avro_schema_compatible(Reader, Writer)
+  when ?IS_ENUM_TYPE(Reader) andalso ?IS_ENUM_TYPE(Writer) ->
+  avro:is_same_type(Reader, Writer) andalso
+    (length(Writer#avro_enum_type.symbols --
```

Comment:
nit: `Writer#avro_enum_type.symbols -- Reader#avro_enum_type.symbols == []` instead of checking its length.